managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedCursorContainer.java

```diff
@@ -56,10 +76,61 @@ private static class Item {
         }
     }
 
-    public ManagedCursorContainer() {
+    /**
+     * Utility class to manage a data version, which rolls over to 0 when reaching Long.MAX_VALUE.
+     */
+    public static final class DataVersion {
+        private DataVersion() {}
+
+        /**
+         * Compares two data versions, which either rolls overs to 0 when reaching Long.MAX_VALUE.
+         * <p>
```

Comment:
I clarified by adding:
```
         * Use {@link DataVersion#incrementVersion(long)} to increment the versions. The assumptions
         * is that metric versios are compared with close time proximity one to another, hence,
         * they are expected not close to each other in terms of distance, hence we don't
         * expect the distance ever to exceed Long.MAX_VALUE / 2, otherwise we wouldn't be able
         * to know which one is a later version in case the furthest rolls over to beyond 0. We
         * assume the shortest distance between them dictates that.
```